File: backend/app/schemas/api_response.py
"""공통 API 응답 계약을 정의한다.

모든 API 응답이 공유하는 성공 여부, 데이터, 메시지 구조를 제공한다.
"""

from typing import Generic, TypeVar

from pydantic import BaseModel


response_data_type = TypeVar("response_data_type")


class ApiResponse(BaseModel, Generic[response_data_type]):
    """모든 API Endpoint에서 사용하는 공통 응답 구조이다."""

    success: bool
    data: response_data_type | None = None
    message: str | None = None

File: frontend/types/api_response.ts
// 목적: Backend 공통 API 응답 계약(ApiResponse)의 Frontend 대응 타입을 정의한다.
// 주요 역할: Backend의 success/data/message 구조를 그대로 미러링해 API 연동 담당자가
//           재사용할 수 있게 한다. API JSON 필드는 snake_case 규칙을 유지한다.

export interface ApiResponse<T> {
  success: boolean;
  data: T | null;
  message: string | null;
}
